File: scalechain-wallet/src/test/scala/io/scalechain/wallet/WalletStoreSpec.scala
package io.scalechain.wallet

import java.io.File

import io.scalechain.blockchain.storage.Storage
import io.scalechain.blockchain.storage.index.RocksDatabase
import io.scalechain.blockchain.transaction.ChainEnvironmentFactory
import org.apache.commons.io.FileUtils
import org.scalatest._

import scala.util.Random

class WalletStoreSpec extends FlatSpec with BeforeAndAfterEach with ShouldMatchers
  //with WalletStoreAccountTestTrait
  with WalletStoreOutPointTestTrait
  with WalletStoreTransactionHashTestTrait
  with WalletStoreWalletOutputTestTrait // Need to fix the protocol codec exception.
  with WalletStoreWalletTransactionTestTrait
  {
  this: Suite =>

  var store : WalletStore = null

  if (!Storage.isInitialized)
    Storage.initialize()

  if (ChainEnvironmentFactory.getActive().isEmpty)
    ChainEnvironmentFactory.create("testnet")

  val testPath = new File(s"./target/unittests-WalletStoreSpec")

  override def beforeEach() {
    println("BeforeEach WalletStoreSpec")

    println("deleting files...")

    FileUtils.deleteDirectory( testPath )
    store = new WalletStore( testPath )

    super.beforeEach()
  }

  override def afterEach() {
    super.afterEach()

    store.close()
    FileUtils.deleteDirectory( testPath )
    store = null
  }
}

File: scalechain-wallet/src/test/kotlin/io/scalechain/wallet/WalletStoreSpec.kt
package io.scalechain.wallet

import io.kotlintest.KTestJUnitRunner
import java.io.File

import io.scalechain.blockchain.storage.Storage
import io.scalechain.blockchain.transaction.ChainTestTrait
import org.junit.runner.RunWith
import java.util.*

import scala.util.Random

@RunWith(KTestJUnitRunner::class)
class WalletStoreSpec : WalletTestTrait(), WalletTests {

  override val testPath = File("./target/unittests-WalletStoreSpec-${Random().nextLong()}")

  lateinit override var store : WalletStore

  init {
    if (!Storage.isInitialized)
      Storage.initialize()

    addTests()
  }

  override fun beforeEach() {
    super.beforeEach()

    store = WalletStore()
  }

  override fun afterEach() {
    super.afterEach()
  }
}
